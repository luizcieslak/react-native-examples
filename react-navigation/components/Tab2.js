import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

const Tab2 = ({ navigation }) => {
  return (
    <View style={styles.container}>
      <Text style={styles.title}>Tab2</Text>
      <Text style={styles.subtitle}>Page2</Text>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  title: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  subtitle: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
});

export default Tab2;