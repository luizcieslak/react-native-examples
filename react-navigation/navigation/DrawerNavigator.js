import React from 'react'
import Image from 'react-native'
import { createDrawerNavigator } from 'react-navigation'

import Drawer1 from '../components/Drawer1'
import Drawer2 from '../components/Drawer2'

export default createDrawerNavigator({
  Drawer1: {
    screen: Drawer1
  },
  Drawer2: {
    screen: Drawer2
  }
})