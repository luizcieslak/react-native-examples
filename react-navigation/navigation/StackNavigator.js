import React from 'react';
import { createStackNavigator } from 'react-navigation'

import Stack1 from '../components/Stack1'
import Stack2 from '../components/Stack2'
import LogoTitle from '../components/LogoTitle';

export default createStackNavigator({
  Stack1: Stack1,
  Stack2: Stack2,
}, {
  defaultNavigationOptions: ({ navigation }) => ({
    headerTintColor: '#000',
    headerStyle: {
      backgroundColor: '#fff',
    },
    //title: navigation.state.routeName
    headerTitle: <LogoTitle title={navigation.state.routeName} />
  }),
})