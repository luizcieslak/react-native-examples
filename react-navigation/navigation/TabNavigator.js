import { createBottomTabNavigator, createAppContainer } from 'react-navigation'
//import screens
import Tab1 from '../components/Tab1'
import Tab2 from '../components/Tab2'
import Tab3 from '../components/Tab3'

const TabNavigator = createBottomTabNavigator({
  Tab1: Tab1,
  Tab2: Tab2,
  Tab3: Tab3,
})

export default createAppContainer(TabNavigator)