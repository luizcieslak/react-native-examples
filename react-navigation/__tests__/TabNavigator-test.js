import React from 'react'
import renderer from 'react-test-renderer';
import TabNavigator from '../navigation/TabNavigator'
//import screens
import StackNavigator from '../navigation/StackNavigator'
import Tab2 from '../components/Tab2'
import DrawerNavigator from '../navigation/DrawerNavigator'

// Mock react-navigation functions
jest.mock('react-navigation', () => {
  return {
    createAppContainer: jest.fn().mockReturnValue(function NavigationContainer(props) { return null; }),
    createDrawerNavigator: jest.fn(),
    createBottomTabNavigator: jest.fn(),
    createMaterialTopTabNavigator: jest.fn(),
    createStackNavigator: jest.fn(),
    StackActions: {
      push: jest.fn().mockImplementation(x => ({ ...x, "type": "Navigation/PUSH" })),
      replace: jest.fn().mockImplementation(x => ({ ...x, "type": "Navigation/REPLACE" })),
    },
    NavigationActions: {
      navigate: jest.fn().mockImplementation(x => x),
    }
  }
});

test('TabNavigation renders correctly', () => {
  const tree = renderer.create(<TabNavigator />).toJSON()
  expect(tree).toMatchSnapshot();
})