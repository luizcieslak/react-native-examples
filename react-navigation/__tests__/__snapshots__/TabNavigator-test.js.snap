// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabNavigation renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "overflow": "hidden",
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      collapsable={false}
      pointerEvents="auto"
      removeClippedSubviews={false}
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "opacity": 1,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          onLayout={[Function]}
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            collapsable={false}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            style={
              Object {
                "backgroundColor": "#eee",
                "flex": 1,
                "flexDirection": "column-reverse",
                "overflow": "hidden",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                pointerEvents="auto"
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "paddingTop": 64,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  accessibilityElementsHidden={false}
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "flex": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#F5FCFF",
                        "flex": 1,
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "fontSize": 20,
                          "margin": 10,
                          "textAlign": "center",
                        }
                      }
                    >
                      Stack1
                    </Text>
                    <Text
                      style={
                        Object {
                          "color": "#333333",
                          "marginBottom": 5,
                          "textAlign": "center",
                        }
                      }
                    >
                      Page1
                    </Text>
                    <View
                      accessibilityRole="button"
                      accessibilityStates={Array []}
                      accessible={true}
                      isTVSelectable={true}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "opacity": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {},
                          ]
                        }
                      >
                        <Text
                          style={
                            Array [
                              Object {
                                "color": "#007AFF",
                                "fontSize": 18,
                                "padding": 8,
                                "textAlign": "center",
                              },
                            ]
                          }
                        >
                          Next
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View
              onLayout={[Function]}
              pointerEvents="box-none"
              style={
                Object {
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#fff",
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  onLayout={[Function]}
                  pointerEvents="box-none"
                  style={
                    Object {
                      "backgroundColor": "#fff",
                      "borderBottomColor": "#A7A7AA",
                      "borderBottomWidth": 0.5,
                      "height": 64,
                      "paddingBottom": 0,
                      "paddingLeft": 0,
                      "paddingRight": 0,
                      "paddingTop": 20,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "bottom": 0,
                          "flexDirection": "row",
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                        }
                      }
                    >
                      <View
                        pointerEvents="box-none"
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "transparent",
                            "bottom": 0,
                            "flexDirection": "row",
                            "justifyContent": "center",
                            "left": 0,
                            "opacity": 1,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "flex": 1,
                              "flexDirection": "row",
                            }
                          }
                        >
                          <Image
                            source={
                              Object {
                                "testUri": "../../../assets/rn-logo.png",
                              }
                            }
                            style={
                              Object {
                                "height": 50,
                                "marginLeft": 10,
                                "width": 50,
                              }
                            }
                          />
                          <Text
                            style={
                              Object {
                                "alignSelf": "center",
                                "fontSize": 25,
                                "marginLeft": 5,
                              }
                            }
                          >
                            Stack1
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      Object {
        "backgroundColor": "#fff",
        "borderTopColor": "rgba(0, 0, 0, .3)",
        "borderTopWidth": 0.5,
        "flexDirection": "row",
        "height": 49,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
      }
    }
  >
    <View
      accessibilityLabel="Tab1, tab, 1 of 3"
      accessible={true}
      hitSlop={
        Object {
          "bottom": 5,
          "left": 15,
          "right": 15,
          "top": 5,
        }
      }
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
          },
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "flexDirection": "column",
            "justifyContent": "flex-end",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 29,
            },
            false,
            Object {
              "flex": 1,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "height": "100%",
              "justifyContent": "center",
              "minWidth": 25,
              "opacity": 1,
              "position": "absolute",
              "width": "100%",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "black",
                  "fontSize": 25,
                },
                undefined,
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "height": "100%",
              "justifyContent": "center",
              "minWidth": 25,
              "opacity": 0,
              "position": "absolute",
              "width": "100%",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "gray",
                  "fontSize": 25,
                },
                undefined,
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <Text
        allowFontScaling={true}
        numberOfLines={1}
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "black",
            "fontSize": 11,
            "marginBottom": 1.5,
            "textAlign": "center",
          }
        }
      >
        Tab1
      </Text>
    </View>
    <View
      accessibilityLabel="Tab2, tab, 2 of 3"
      accessible={true}
      hitSlop={
        Object {
          "bottom": 5,
          "left": 15,
          "right": 15,
          "top": 5,
        }
      }
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
          },
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "flexDirection": "column",
            "justifyContent": "flex-end",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 29,
            },
            false,
            Object {
              "flex": 1,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "height": "100%",
              "justifyContent": "center",
              "minWidth": 25,
              "opacity": 0,
              "position": "absolute",
              "width": "100%",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "black",
                  "fontSize": 25,
                },
                undefined,
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "height": "100%",
              "justifyContent": "center",
              "minWidth": 25,
              "opacity": 1,
              "position": "absolute",
              "width": "100%",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "gray",
                  "fontSize": 25,
                },
                undefined,
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <Text
        allowFontScaling={true}
        numberOfLines={1}
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "gray",
            "fontSize": 11,
            "marginBottom": 1.5,
            "textAlign": "center",
          }
        }
      >
        Tab2
      </Text>
    </View>
    <View
      accessibilityLabel="Tab3, tab, 3 of 3"
      accessible={true}
      hitSlop={
        Object {
          "bottom": 5,
          "left": 15,
          "right": 15,
          "top": 5,
        }
      }
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
          },
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "flexDirection": "column",
            "justifyContent": "flex-end",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 29,
            },
            false,
            Object {
              "flex": 1,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "height": "100%",
              "justifyContent": "center",
              "minWidth": 25,
              "opacity": 0,
              "position": "absolute",
              "width": "100%",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "black",
                  "fontSize": 25,
                },
                undefined,
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "height": "100%",
              "justifyContent": "center",
              "minWidth": 25,
              "opacity": 1,
              "position": "absolute",
              "width": "100%",
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "gray",
                  "fontSize": 25,
                },
                undefined,
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
      <Text
        allowFontScaling={true}
        numberOfLines={1}
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "gray",
            "fontSize": 11,
            "marginBottom": 1.5,
            "textAlign": "center",
          }
        }
      >
        Tab3
      </Text>
    </View>
  </View>
</View>
`;
